storage/isolation_manager.rs

```diff
@@ -104,8 +104,8 @@ impl IsolationManager {
         let window = self.timeline.get_or_create_window(sequence_number);
         window.insert_pending(sequence_number, commit_record);
         let CommitStatus::Pending(commit_record) = window.get_status(sequence_number) else { unreachable!() };
-        let isolation_conflict = self.validate_all_concurrent(sequence_number, &commit_record, durability_client)?;
-        if isolation_conflict.is_none() {
+        let validation_result = self.validate_all_concurrent(sequence_number, &commit_record, durability_client);
```

Comment:
Oof this is suuper subtle. That's actually why I went towards https://github.com/typedb/typedb/pull/7664/files - would you have a look at that idea?